Dockerfile

```diff
@@ -1,5 +1,6 @@
 # Build the manager binary
-FROM golang:1.23 AS builder
+ARG GO_VERSION
```

Comment:
Do we want to set a default value there to prevent the warning?
https://docs.docker.com/reference/dockerfile/#default-values

```
 1 warning found (use docker --debug to expand):
 - InvalidDefaultArgInFrom: Default value for ARG golang:${GO_VERSION} results in empty or invalid base image name (line 3)
```

Even if we expect to always build the image via `make`, which sets a default value, it probably wouldn't hurt to set a default here too, WDYT?

Reply:
Yeah that makes sense. I'll update and probably push a PR to CAPI/CAPO also since they both get the same warning